pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -739,3 +739,28 @@ func deleteJob(cj *batchv1.CronJob, job *batchv1.Job, jc jobControlInterface, re
 func getRef(object runtime.Object) (*corev1.ObjectReference, error) {
 	return ref.GetReference(scheme.Scheme, object)
 }
+
+func getString(s *string) string {
+	if s == nil {
+		return ""
+	}
+	return *s
+}
+
+func formatSchedule(timeZoneEnabled bool, cj *batchv1.CronJob, recorder record.EventRecorder) string {
+	if strings.Contains(cj.Spec.Schedule, "TZ") {
+		recorder.Eventf(cj, corev1.EventTypeWarning, "UnsupportedSchedule", "CRON_TZ or TZ used in schedule %q is not officially supported, see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/ for more details", cj.Spec.Schedule)
+		return cj.Spec.Schedule
+	}
+
+	if timeZoneEnabled && cj.Spec.TimeZone != nil {
+		if _, err := time.LoadLocation(*cj.Spec.TimeZone); err != nil {
+			recorder.Eventf(cj, corev1.EventTypeWarning, "UnknownTimeZone", "%v, using apiserver's local time instead")
```

Comment:
The thought was to fallback to the pre-timeZone field functionality of using local time. This behavior emits a warning explaining what's happening. Also, if an invalid timeZone is specified, it will fail validation.

Reply:
> Also, if an invalid timeZone is specified, it will fail validation.

not if the API object was created against a 1.X API server on a new version of golang that knows about a new named time zone, and kube-controller-manager is still at 1.(X-1) built on an older version of golang that doesn't understand that time zone

Reply:
Probably emitting the warning and suspending the job in the cases where as Jordan mentions this happened during update is the best possible option. Also the local time zone is from kube-controller-manager, not from the apiserver, since that's where the cronjob controller runs.